test/libdnf5/conf/test_conf.cpp

```diff
@@ -71,3 +71,57 @@ void ConfTest::test_config_pkg_gpgcheck() {
     CPPUNIT_ASSERT_EQUAL(&config_repo.get_pkg_gpgcheck_option(), &config_repo.get_gpgcheck_option());
 #pragma GCC diagnostic pop
 }
+
+libdnf5::ConfigMain copy_construct_config_main(const libdnf5::ConfigMain & config) {
+    return libdnf5::ConfigMain{config};
+}
+
+libdnf5::ConfigMain assign_config_main(const libdnf5::ConfigMain & config) {
+    libdnf5::ConfigMain config_copy;
+    config_copy = config;
+    return config_copy;
```

Comment:
Hmm. NRVO (Named Return Value Optimization) is an optional optimization, so the object might be copied and then moved. In the absence of a move constructor, two copies would indeed occur. However, modern compilers usually perform NRVO. Moreover, If the code works with a copy followed by a move (or two copies), it will certainly pass the test with a single copy. Therefore, the test's logic is sound.